docs/source/tutorial/interactions.rst

```diff
@@ -0,0 +1,485 @@
+interpreting concatenated embedding vectors as image, pairs of embedding vectors as normal distributions, or semantic
+matching inspired sums of linear products.
+
+Moreover, some choose a form that can be decomposed into
+
+.. math::
+
+    f(\mathbf{h}, \mathbf{r}, \mathbf{t}) = f_o(f_i(\mathbf{h}, \mathbf{r}), \mathbf{t})
+
+with an expensive $f_i$ and a cheap $f_o$. Such form allows efficient scoring of many tails for a given head-relation
+combination, and can becombined with inverse relation modelling for an overall efficient training and inference
+architecture.
+
+ConvE
+~~~~~
+
+:class:`~pykeen.nn.modules.ConvEInteraction` uses an interaction of the form
+
+.. math::
+
+    \langle g(\mathbf{h}, \mathbf{r}), \mathbf{t} \rangle + t_b
+
+for $\mathbf{h}, \mathbf{r}, \mathbf{t} \in \mathbb{R}^d$ are the head entity, relation, and tail entity representation,
+and $t_b \in \mathbb{R}$ is an entity bias. $g$ is a CNN-based encoder, which first operates on a 2D-reshaped "image"
+and then flattens the output for a second linear layer. Dropout and batch normalization is utilized, too.
+
+ConvKB
+~~~~~~
+
+:class:`~pykeen.nn.modules.ConvKBInteraction` concatenates $\mathbf{h}, \mathbf{r}, \mathbf{t} \in \mathbb{R}^d$ to a $3
+\times d$ "image" and applies a $3 \times 1$ convolution. The output is flattened and a linear layer predicts the score.
+
+CrossE
+~~~~~~
+
+:class:`~pykeen.nn.modules.CrossEInteraction` uses
+
+.. math::
+
+    \langle g(\mathbf{h}, \mathbf{r}), \mathbf{t} \rangle
+
+where
+
+.. math::
+
+    g(\mathbf{h}, \mathbf{r}) = \sigma(
+        \mathbf{c}_r \odot \mathbf{h}
+        + \mathbf{c}_r \odot \mathbf{h} \odot \mathbf{r}
+        + \mathbf{b}
+    )
+
+with an activation function $\sigma$. Moreover, dropout is applied to the output of $g$.
+
+ERMLP
+~~~~~
+
+:class:`~pykeen.nn.modules.ERMLPInteraction` uses a simple 2-layer MLP on the concatenated head, relation, and tail
+representations $\mathbf{h}, \mathbf{r}, \mathbf{t} \in \mathbb{R}^d$.
+
+ERMLP (E)
+~~~~~~~~~
+
+:class:`~pykeen.nn.modules.ERMLPEInteraction` adjusts :class:`~pykeen.nn.modules.ERMLPInteraction` for a more efficient
+training and inference architecture by using
+
+.. math::
+
+    \langle g(\mathbf{h}, \mathbf{r}), \mathbf{t} \rangle
+
+where $g$ is a 2-layer MLP.
+
+KG2E
+~~~~
+
+:class:`~pykeen.nn.modules.KG2EInteraction` interprets pairs of vectors $\mathbf{h}_{\mu}, \mathbf{h}_{\Sigma},
+\mathbf{r}_{\mu}, \mathbf{r}_{\Sigma}, \mathbf{t}_{\mu}, \mathbf{t}_{\Sigma} \in \mathbb{R}^d$ as normal distributions
+$\mathcal{N}_h, \mathcal{N}_r, \mathcal{N}_t$ and determines a similarity between $\mathcal{N}_h - \mathcal{N}_t$ and
+$\mathcal{N}_r$.
+
+.. todo:: This does not really fit well into the neural category.
+
+Neural Tensor Network (NTN)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:class:`~pykeen.nn.modules.NTNInteraction` defines the interaction function as
+
+.. math::
+
+    \left \langle
+        \mathbf{r}_{u},
```

Comment:
what is u?

Reply:
A relation-specific parameter :shrug: :smile: 

Everything with `r` or `R` is relation-specific. NTN uses 4 different tensors for each relation.

Reply:
Okay. It’s just a bit confusing with a letter than has no text explanation. Maybe just update the text to say so. :)

Reply:
We have this part:

> $\mathbf{r}_1, \mathbf{r}_u \in \mathbb{R}^d, \mathbf{R}_2 \in \mathbb{R}^{k \times 2d}, \mathbf{R}_3 \in \mathbf{R}^{d \times d \times k}$ are relation-specific parameters

Reply:
I know :) I just don't understand why it's r_1 and r_u instead of r_2. Like what does u mean??? I guess the answer is they already went 1, 2, 3 with the other things, then added this u since 4 would have been weird